relay-general/src/pii/processor.rs

```diff
@@ -945,3 +954,27 @@ fn test_ip_address_hashing_does_not_overwrite_id() {
 
     assert_eq!(user.id.value().unwrap().as_str(), "123");
 }
+
+#[test]
+fn test_replace_replaced_text() {
```

Comment:
nit: It might be nice to have an additional test with an actual payload example (something like this): https://github.com/getsentry/relay/blob/358014e6fc61f01d53ec02468160cc37db400e7c/relay-general/src/pii/convert.rs#L1136-L1146 to prove that the issue is now fixed.